packages/server/api/src/app/block-variable/block-variable-module.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  FastifyPluginAsyncTypebox,
+  Type,
+} from '@fastify/type-provider-typebox';
+import {
+  encodeStepOutputs,
+  EndpointScope,
+  flowHelper,
+  PrincipalType,
+} from '@openops/shared';
+import { engineRunner } from 'server-worker';
+import { accessTokenManager } from '../authentication/lib/access-token-manager';
+import { flowService } from '../flows/flow/flow.service';
+import { flowStepTestOutputService } from '../flows/step-test-output/flow-step-test-output.service';
+
+export const blockVariableModule: FastifyPluginAsyncTypebox = async (app) => {
+  await app.register(blockVariableController, { prefix: '/v1/block-variable' });
+};
+
+const blockVariableController: FastifyPluginAsyncTypebox = async (app) => {
+  app.post('/execute-variable', ExecuteVariableRequest, async (req) => {
+    const request = req.body;
+    const { projectId } = req.principal;
+
+    const flow = await flowService.getOnePopulatedOrThrow({
+      projectId,
+      id: request.flowId,
+      versionId: request.flowVersionId,
+    });
+
+    const engineToken = await accessTokenManager.generateEngineToken({
+      projectId,
+    });
+
+    const stepIds = flowHelper.getAllStepIds(flow.version.trigger);
+    const outputs = await flowStepTestOutputService.listEncrypted({
+      flowVersionId: request.flowVersionId,
+      stepIds,
+    });
+
+    const stepTestOutputs = encodeStepOutputs(outputs);
+
+    const { result } = await engineRunner.executeVariable(engineToken, {
+      variableExpression: request.variableExpression,
+      flowVersion: flow.version,
+      projectId,
+      stepName: request.stepName ?? '',
+      stepTestOutputs,
+    });
+
+    return {
+      success: result.success,
+      value: result.censoredValue,
+      error: result.error,
+    };
```

Comment:
th enginer returnes both r`esolvedValue` and `censoredValue`, so we just return the censored one